splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java

```diff
@@ -574,72 +588,156 @@ private ConflictResults checkCommitTimestampForConflict(TxnView updateTransactio
+                        if (conflictingTxns != null) {
+                            if(LOG.isTraceEnabled()) {
+                                LOG.trace(String.format("adding txn(s) %s to the list of txn %d's conflicting txns",
+                                                        Arrays.toString(conflictingTxns),
+                                                        updateTransaction.getTxnId()));
+                            }
+                            txnStore.addConflictingTxnIds(updateTransaction.getTxnId(), conflictingTxns);
+                        }
                     }
-                    conflictResults.addAdditive(dataTransactionId);
                     break;
-                case SIBLING:
-                    if(LOG.isTraceEnabled()){
-                        SpliceLogUtils.trace(LOG,"Write conflict on row "
-                                +Bytes.toHex(cell.keyArray(),cell.keyOffset(),cell.keyLength()));
-                    }
-                    throw exceptionLib.writeWriteConflict(dataTransactionId,updateTransaction.getTxnId());
-            }
+                }
         }
+
         return conflictResults;
     }
 
     // Helpers
 
+    private long[] getDataActiveTransactions(TxnView txn, Partition table, KVPair kvPair, TxnSupplier txnSupplier, DataCell cell) throws IOException {
+        if(shouldIgnoreConflictsInternal(table)) {
+            if(LOG.isTraceEnabled()) {
+                LOG.trace(String.format("detection of conflicting data active txns ignored for txn %d as it affects table %s",
+                                        txn.getTxnId(),
+                                        table.getTableName()));
+            }
+            return null;
+        }
+        DataGet dataGet = opFactory.newGet(kvPair.getRowKey(), null);
+        dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.TOMBSTONE_COLUMN_BYTES);
+        dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.PACKED_COLUMN_BYTES);
+        if(kvPair.getType() == KVPair.Type.DELETE) {
+            dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.FK_COUNTER_COLUMN_BYTES);
+        }
+        dataGet.returnAllVersions();
+        DataResult result = table.get(dataGet, null);
+        Iterator<DataCell> iterator = result.iterator();
+        Set<Long> activeTx = new HashSet<>();
+        while(iterator.hasNext()) {
+            long dataTxnId = iterator.next().version();
+            if((dataTxnId & SIConstants.TRANSANCTION_ID_MASK) == txn.getTxnId()) {
+                continue;
+            }
+            // get the parent
+            TxnView txnView = txnSupplier.getTransaction(dataTxnId);
+            long id = txnView.getTxnId();
+            while(true) {
+                TxnView parent = txnView.getParentTxnView();
+                if(parent == Txn.ROOT_TRANSACTION) break;
+                txnView = parent;
+            }
+            Txn.State state = txnView.getState();
```

Comment:
Should we add the '((txnView.getTxnId() & SIConstants.TRANSANCTION_ID_MASK) == txn.getTxnId())' check here and continue, or is it OK to add this transaction's txnId to the HashSet?

Reply:
I cleaned it a bit, the resulting `txnView` should always have its last byte set to zero, added an assertion for that.

Reply:
**update** I was mistaken, I think if we could reach this code path by a in-memory sub-transaction which has its last byte set to its child-order (I think) to the parent. In that case we should reset the last byte in order to get the parent transaction (since we want to rollback the parent transaction if we conflict with a child sub-transaction).